packages/compass-web/sandbox/index.tsx

```diff
@@ -0,0 +1,272 @@
+  all: 'unset',
+  display: 'block',
+  width: '100%',
+  cursor: 'pointer',
+  color: palette.black,
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+});
+
+resetGlobalCSS();
+
+function getHistory(): string[] {
+  try {
+    const b64Str = localStorage.getItem('CONNECTIONS_HISTORY');
+    if (!b64Str) {
+      return [];
+    }
+    const binStr = window.atob(b64Str);
+    const bytes = Uint8Array.from(binStr, (v) => v.codePointAt(0) ?? 0);
+    const str = new TextDecoder().decode(bytes);
+    return JSON.parse(str);
+  } catch (err) {
+    return [];
+  }
+}
+
+function saveHistory(history: any) {
+  try {
+    const bytes = new TextEncoder().encode(JSON.stringify(history));
+    const binStr = String.fromCodePoint(...bytes);
+    const b64Str = window.btoa(binStr);
+    localStorage.setItem('CONNECTIONS_HISTORY', b64Str);
+  } catch (err) {
+    // noop
+  }
+}
+
+function validateConnectionString(str: string) {
+  try {
+    new ConnectionString(str);
+    return null;
+  } catch (err) {
+    return (err as Error).message;
+  }
+}
+
+const App = () => {
+  const [initialTab] = useState<OpenWorkspaceOptions>(() => {
+    const [, tab, namespace = ''] = window.location.pathname.split('/');
+    if (tab === 'databases') {
+      return { type: 'Databases' };
+    }
+    if (tab === 'collections' && namespace) {
+      return { type: 'Collections', namespace };
+    }
+    if (tab === 'collection' && namespace) {
+      return { type: 'Collection', namespace };
+    }
+    return { type: 'Databases' };
+  });
+  const [connectionsHistory, setConnectionsHistory] = useState<string[]>(() => {
+    return getHistory();
+  });
+  const [focused, setFocused] = useState(false);
+  const [connectionString, setConnectionString] = useState('');
+  const [openCompassWeb, setOpenCompassWeb] = useState(false);
+  const [
+    connectionStringValidationResult,
+    setConnectionStringValidationResult,
+  ] = useState<null | string>(null);
+
+  const canSubmit =
+    connectionStringValidationResult === null && connectionString !== '';
+
+  const onChangeConnectionString = useCallback((str: string) => {
+    setConnectionStringValidationResult(validateConnectionString(str));
+    setConnectionString(str);
+  }, []);
+
+  const onConnectClick = useCallback(() => {
+    setOpenCompassWeb(true);
+    setConnectionsHistory((history) => {
+      if (history.includes(connectionString)) {
+        return history;
+      }
+      history.unshift(connectionString);
+      if (history.length > 10) {
+        history.pop();
+      }
+      saveHistory(history);
+      return [...history];
+    });
+  }, [connectionString]);
+
+  if (openCompassWeb) {
+    return (
+      <Body as="div" className={sandboxContainerStyles}>
+        <ErrorBoundary>
+          <CompassWeb
+            connectionString={connectionString}
+            initialWorkspaceTabs={[initialTab]}
+            onActiveWorkspaceTabChange={(tab) => {
```

Comment:
Just an example of how using existing workspace interfaces initial state can be derived from and synced back to the browser history